lib/frontman/app.rb

```diff
@@ -94,11 +94,21 @@ def add_to_manifest(key, value)
       @assets_manifest[key] = '/' + value.sub(%r{^/}, '')
     end
 
-    sig { params(dirs: Array).void }
+    sig { params(dirs: T.any(Array, Hash)).void }
     def register_data_dirs(dirs)
-      dirs.each do |dir|
-        define_singleton_method dir.to_sym do
-          @data_dirs[dir] ||= DataStore.new(File.join(Dir.pwd, dir))
+      if dirs.is_a?(Array)
+        array = dirs.dup
+
+        dirs = {}
+
+        array.each do |dir|
+          dirs[dir.split("/").last] = dir
+        end
```

Comment:
We could probably refactor this to a one-liner, something like this:

```ruby
dirs = dirs.map { |dir| [dir.split("/").last, dir] }.to_h
```

WDYT?